arch/xtensa/src/esp32/esp32_wifi_adapter.c

```diff
@@ -3447,7 +3469,17 @@ static void *esp_malloc_internal(size_t size)
 
 static void *esp_realloc_internal(void *ptr, size_t size)
 {
+#ifdef CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP
+  return xtensa_imm_realloc(ptr, size);
+#else
+  if (size == 0 || esp32_ptr_extram(ptr))
+    {
+      esp_free(ptr);
+      return NULL;
+    }
+
   return kmm_realloc(ptr, size);
```

Comment:
i guess you want to check esp32_ptr_extram on the result of kmm_realloc.

Reply:
Yes, I want to make sure that the old memory pointer is not in external RAM (PSRAM), although this is unlikely to happen.

Reply:
`kmm_realloc ` also needs to be checked by `esp32_ptr_extram`, I have modified it, thank you for your reminder.